lib/api/CDataFrameTrainBoostedTreeClassifierRunner.cc

```diff
@@ -111,6 +112,8 @@ void CDataFrameTrainBoostedTreeClassifierRunner::writeOneRow(
     writer.StartObject();
     writer.Key(this->predictionFieldName());
     writer.String(categoryValues[predictedCategoryId]);
+    writer.Key(PREDICTION_PROBABILITY_FIELD_NAME);
+    writer.Double(probabilityOfCategory[0]);
```

Comment:
What do we call `probabilityOfCategory1` above? It seems a bit confusing.

Reply:
Ooops, at this point the entries are not sorted yet. Let me fix that and add unit test.

Reply:
I've added unit test. In order to do that I needed to split `writeOneRow` method. I don't quite like this but I couldn't find an easy way to mock out the boosted tree.
@tveasey: Any suggestion for improvement here?

As for the naming, I kept the name `probabilityOfCategory1` as I couldn't come up with a better one. I added inline comments so hopefully the meaning of this variable is now clear.